### PR TITLE
Docker tweaks

### PR DESCRIPTION
**Pull request by israelboudoux**
## What does this implement/fix? Explain your changes.
It adds missing env vars to docker-compose files and sorts the env vars alphabetically (considers first letter only).

## Does this close any currently open issues?
No

## What commands can I run to test the change? 
The current in the pipeline

## Any other comments?
No

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -121,18 +130,21 @@ services:
...
       LOG_LEVEL: ${LOG_LEVEL:-debug}
       LOG_HTTP_PAYLOAD_ENABLED: ${LOG_HTTP_PAYLOAD_ENABLED:-true}
       LOG_HTTP_FULL_DATA: ${LOG_HTTP_FULL_DATA:-false}
-      IS_CHALLENGER: ${IS_CHALLENGER:-true}
-      PROPOSER_MAX_BLOCK_PERIOD_MILIS: ${PROPOSER_MAX_BLOCK_PERIOD_MILIS:-0}
       MAX_BLOCK_SIZE: ${MAX_BLOCK_SIZE:-50000}
+      MONGO_URL: ${OPTIMIST_MONGO_URL:-mongodb://mongodb:27017}
+      PROPOSER_MAX_BLOCK_PERIOD_MILIS: ${PROPOSER_MAX_BLOCK_PERIOD_MILIS:-0}
+      USE_EXTERNAL_NODE: ${USE_EXTERNAL_NODE}
+      WEBSOCKET_PORT: 8080
```
Comment (daveroga) [inline docker/docker-compose.yml:147]: We could always pass some variable in parameters or default value as a normal rule in docker compose files.
```
WEBSOCKET_PORT: ${WEBSOCKET_PORT:8080}
```

**Review by daveroga — CHANGES_REQUESTED** [review]

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -67,30 +72,34 @@ services:
     networks:
       - nightfall_network
     environment:
+      ALWAYS_DO_TRUSTED_SETUP: ${ALWAYS_DO_TRUSTED_SETUP}
+      BLOCKCHAIN_URL: ${BLOCKCHAIN_URL}
+      BLOCKCHAIN_WS_HOST: ${BLOCKCHAIN_WS_HOST:-blockchain}
+      BLOCKCHAIN_PORT: ${BLOCKCHAIN_PORT:-8546}
```
Comment (imagobea) [inline docker/docker-compose.yml:78]: `BLOCKCHAIN_PORT` on top of `BLOCKCHAIN_URL` (?)
`DAI_RESTRICT` should be on top of `DEPLOY_`

For me obvs this is nice but from experience alphabetical order is also hard to maintain...
What do you think? Could a comment help perhaps??

**Review by imagobea — APPROVED** [review]
Hello @israelboudoux 

I found some instances that aren't in strict alphabetical order. Like I said in the comments, of course I like the approach, but might be hard to maintain. Therefore, feel free to take another look and apply changes or move on.

Thank you!! :boom:

File: docker/docker-compose.deployment.yml [hunk 3]
```diff
@@ -136,7 +141,6 @@ services:
       PROPOSER_MAX_BLOCK_PERIOD_MILIS: ${PROPOSER_MAX_BLOCK_PERIOD_MILIS:-0}
       USE_EXTERNAL_NODE: ${USE_EXTERNAL_NODE}
       WEBSOCKET_PORT: 8080
-      GAS_ESTIMATE_ENDPOINT: ${GAS_ESTIMATE_ENDPOINT}
```
[the remark this answers is not in the document]
Comment (israelboudoux) [reply docker/docker-compose.deployment.yml:139]: Yes, at least from my search I didn't find it being used by the Optimist. I saw that it is used only by the Nf3 class, which is not used by the Optimist.

File: docker/docker-compose.deployment.yml [hunk 4]
```diff
@@ -69,24 +69,29 @@ services:
       - ./volumes/build:/app/build/
     environment:
       ALWAYS_DO_TRUSTED_SETUP: ${ALWAYS_DO_TRUSTED_SETUP}
-      ENVIRONMENT: ${ENVIRONMENT:?ENVIRONMENT-cannot-be-blank}
       BLOCKCHAIN_URL: ${BLOCKCHAIN_URL:?BLOCKCHAIN_URL-cannot-be-blank}
       CIRCOM_WORKER_HOST: ${CIRCOM_WORKER_HOST:-worker}
       DEPLOY_MOCK_TOKENS: ${DEPLOY_MOCK_TOKENS?DEPLOY_MOCK_TOKENS-not-set}
       DEPLOY_MOCKED_SANCTIONS_CONTRACT: ${DEPLOY_MOCKED_SANCTIONS_CONTRACT:?DEPLOY_MOCKED_SANCTIONS_CONTRACT-cannot-be-blank}
+      DAI_RESTRICT: ${DAI_RESTRICT}
```
[the remark this answers is not in the document]
Comment (israelboudoux) [reply docker/docker-compose.deployment.yml:76]: I just sorted out by the first letter :smiley:

File: docker/docker-compose.yml (hunk 1 above)
Comment (israelboudoux) [reply docker/docker-compose.yml:147]: Good catch, this was a copy/paste thing.

File: docker/docker-compose.yml (hunk 2 above)
Comment (imagobea) [reply docker/docker-compose.yml:78]: Completely missed the "first letter" thing :upside_down_face:

**Review by daveroga — APPROVED** [review]

**Pull request merged**